### ISSUE TITLE
Heredoc highlighting recognizes invalid heredoc-terminators

### ISSUE THREAD

**Issue by ljw1004** [atom/language-php#277]
### Description

The PHP spec says that if a heredoc starts with `<<<EOT` then the only valid terminator is `EOT` or `EOT;`. It is **not** a valid terminator if anything other than a single semicolon follows the `EOT`.

But Atom's php syntax highlighter believes that any line which starts with `EOT` followed by a non-alphanumeric is a heredoc-terminator.

This might seem like a niche unimportant edge-case but it has a real-world scenario. I write `EOT.$bar.<<<EOF` expecting to terminate one heredoc then insert a variable then start a new heredoc -- and the syntax highlighter tells me that what I'm doing is valid -- but then the runtime behavior shows that I was mistaken.

### Steps to reproduce

Write a PHP file with the following contents:

```php
$foo = <<<'EOF'
$bar and this and that
EOF.$bar.<<<'EOF'
and some more
EOF;
```

![image](https://user-images.githubusercontent.com/3316258/30757375-58df2c8a-9f83-11e7-94fa-420db3b073c7.png)

*** **Actual behavior:** the middle line `EOF.$bar.<<<'EOF'` is colored as if it terminates one heredoc, has a variable, then starts a new heredoc.

*** **Expected behavior:** The middle line `EOF.$bar.<<<'EOF'` should all be colored green because it all just a verbatim string, part of the heredoc. It does not terminate the first heredoc then start a new heredoc as you might think.

The relevant PHP spec for heredoc explains "It is very important to note that the line with the closing identifier **must contain no other characters**, except a semicolon (;)". The middle line doesn't meet this restriction and hence isn't a terminator.

And PHP parsers do indeed obey the spec in this respect: http://phpfiddle.org/lite/code/g2xf-edyv

So the syntax-highlighter should too!

**Reproduces how often:** consistently all the time.

### Prerequisites

* [X] Put an X between the brackets on this line if you have done all of the following:
    * Reproduced the problem in Safe Mode: http://flight-manual.atom.io/hacking-atom/sections/debugging/#using-safe-mode
    * Followed all applicable steps in the debugging guide: http://flight-manual.atom.io/hacking-atom/sections/debugging/
    * Checked the FAQs on the message board for common solutions: https://discuss.atom.io/c/faq
    * Checked that your issue isn't already filed: https://github.com/issues?utf8=✓&q=is%3Aissue+user%3Aatom
    * Checked that there is not already an Atom package that provides the described functionality: https://atom.io/packages

### Versions

```
$ atom --version
Atom    : 1.19.7
Electron: 1.6.9
Chrome  : 56.0.2924.87
Node    : 7.4.0

$ apm --version
apm  1.18.4
npm  3.10.10
node 6.9.5 x64
python 2.7.10
git 2.13.5

OS: macOS Sierra 10.12.6
```

### Additional Information

The same bug repros in VSCode: https://github.com/Microsoft/vscode/issues/34851

**Comment by 50Wliu** [comment]
Fixed by #278.

**Comment by ljw1004** [comment]
Wow that was fast! Thank you @50Wliu

**Comment by 50Wliu** [comment]
It was a pretty simple fix :smile:.  Should be fixed in Atom 1.22.0 when it's released.
